eth-contracts/Makefile

```diff
@@ -0,0 +1,7 @@
+test:
+	docker run --rm -p 8545:8545 tockb/geth-priv:v1.9.23 &
+	./wait-for-it.sh -t 0 127.0.0.1:8545 -- echo "node is up"
+	node scripts/waitForMiner.js
+	npx hardhat test --network geth
+
+.PHONY: test
```

Comment:
1. There is `-d` flag for docker to run in daemon mode.
2. This script does not clean after the test.
3. The docker can be reuse among tests, should split it to support reuse it.

try this:

```
test:
	npx hardhat test --network geth

integration-test: start-test-node test clean-test-node

start-test-node:
	docker run --name eth-contracts-test-node -d -p 8545:8545 tockb/geth-priv:v1.9.23
	./wait-for-it.sh -t 0 127.0.0.1:8545 -- echo "node is up"
	node scripts/waitForMiner.js

clean-test-node:
	docker rm -f eth-contracts-test-node

.PHONY: test
```

Reply:
Also, there is no dag cache here?